test/parallel/test-stream-writable-invalid-chunk.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const common = require('../common');
+const stream = require('stream');
+
+function testWriteType(val, objectMode, code) {
+  const writable = new stream.Writable({
+    objectMode,
+    write: () => {}
+  });
+  if (!code) {
+    writable.on('error', common.mustNotCall());
+  } else {
+    writable.on('error', common.expectsError({
+      code: code,
+    }));
+  }
+  writable.write(val);
+}
+
+testWriteType([], false, 'ERR_INVALID_ARG_TYPE');
+testWriteType({}, false, 'ERR_INVALID_ARG_TYPE');
+testWriteType(0, false, 'ERR_INVALID_ARG_TYPE');
+testWriteType(true, false, 'ERR_INVALID_ARG_TYPE');
+testWriteType(0.0, false, 'ERR_INVALID_ARG_TYPE');
+testWriteType(undefined, false, 'ERR_INVALID_ARG_TYPE');
```

Comment:
Not sure how `undefined` should actually be handled, is the current behaviour correct?

Reply:
This is correct, in binary mode `undefined` is invalid.